### PR TITLE
Make SDK buildable in VS 2017

### PR DESCRIPTION
**Pull request by antkmsft**
This change was extracted from #1013, in order to isolate some of the changes.

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]

File: sdk/core/azure-core/src/http/body_stream.cpp [hunk 1]
```diff
@@ -53,7 +54,7 @@ std::vector<uint8_t> BodyStream::ReadToEnd(Context const& context, BodyStream& b
 
   for (auto chunkNumber = 0;; chunkNumber++)
   {
-    buffer.resize((chunkNumber + 1) * chunkSize);
+    buffer.resize((static_cast<decltype(buffer)::size_type>(chunkNumber) + 1) * chunkSize);
```
Comment (ahsonkhan) [inline sdk/core/azure-core/src/http/body_stream.cpp:57]: Why is this necessary and what is this cast doing exactly?

File: sdk/core/azure-core/src/http/logging_policy.cpp [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #include <chrono>
 #include <sstream>
 
-using namespace Azure::Core;
+using Azure::Core::Context;
```
Comment (ahsonkhan) [inline sdk/core/azure-core/src/http/logging_policy.cpp:10]: @CaseyCarter - the guidelines don't mention anything about this, so I am wondering, is it fine to use using statement declarations like this?

**Review by ahsonkhan — COMMENTED** [review]
Looks good.

File: sdk/core/azure-core/src/http/body_stream.cpp (hunk 1 above)
Comment (antkmsft) [reply sdk/core/azure-core/src/http/body_stream.cpp:57]: We remove a warning, stating our intentions more clearly. `resize()` takes `vector<uint8_t>::size_type`, which is defined by implementation, but realistically is `uint64`. `chunkNumber` is `int`.
